### PR TITLE
Update Intel RealSense team and repositories

### LINKED ISSUE #725: Update release team membership
# Updates to realsense_team membership.

## New team members

* [@ashrafk93](https://github.com/IntelRealSense/librealsense/pull/13879)
* [@Nir-Az](https://github.com/IntelRealSense/librealsense/pull/13879)

## Note

i am planning to handle the rolling + all other distro for intel Librealsense releases

previous PR:
https://github.com/ros2-gbp/ros2-gbp-github-org/pull/297/files

### PR DESCRIPTION
**Pull request by SamerKhshiboun**
Resolves #725.

### CODE REVIEW

**Comment by Nir-Az** [comment]
@nuclearsandwich can someone please help with reviewing / merging this PR.
We have a release pending for it and we may loss the sync..

Thanks :)

**Comment by nuclearsandwich** [comment]
@Nir-Az thanks for the ping and my apologies for the delay.

Are you changing the names of the release repositories here or attempting to remove/retire old releases and start new ones?
Changing release repo names requires some extra support since the terraform scripts do not track renames directly.

File: realsense.tf [hunk 1]
```diff
@@ -1,10 +1,12 @@
 locals {
   realsense_team = [
-    "doronhi",
+    "SamerKhshiboun",
+    "IntelRealSense",
+    "Nir-Az"
```
Comment (nuclearsandwich) [inline realsense.tf]: In order to change team members, can you please provide links to the relevant package.xml `<maintainer>` tags in the source repository for these individuals or otherwise provide a link demonstrating that they are authorized maintainers/releasers of these ROS packages.

File: realsense.tf [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
   ]
   realsense_repositories = [
-    "librealsense-release",
-    "ros2_intel_realsense-release",
+    "librealsense2-release",
+    "realsense-ros-release",
```
Comment (nuclearsandwich) [inline realsense.tf:9]: Are these "renames" of the current release repositories or the addition of two repositories and the removal of two others?

Release repositories cannot be deleted so the old repositories must be archived rather than removed.

**Review by nuclearsandwich — CHANGES_REQUESTED** [review]

File: realsense.tf (hunk 2 above)
Comment (SamerKhshiboun) [reply realsense.tf:9]: > Are these "renames" of the current release repositories or the addition of two repositories and the removal of two others?
> 
> Release repositories cannot be deleted so the old repositories must be archived rather than removed.

Renaming of the old release repositories. The new ones links:
https://github.com/IntelRealSense/librealsense2-release
https://github.com/IntelRealSense/realsense-ros-release

File: realsense.tf (hunk 1 above)
Comment (SamerKhshiboun) [reply realsense.tf]: Package 1 - realsense2_camera_msgs
https://github.com/IntelRealSense/realsense-ros/blob/27cc1001e8d72bb11f69af9838f8884a9069a8cc/realsense2_camera_msgs/package.xml#L7

Package 2 - realsense2_description
https://github.com/IntelRealSense/realsense-ros/blob/27cc1001e8d72bb11f69af9838f8884a9069a8cc/realsense2_description/package.xml#L7

Package 3 - realsense2_camera
https://github.com/IntelRealSense/realsense-ros/blob/27cc1001e8d72bb11f69af9838f8884a9069a8cc/realsense2_camera/package.xml#L7

Package 4 - librealsense2
https://github.com/IntelRealSense/librealsense/blob/8ffb17b027e100c2a14fa21f01f97a1921ec1e1b/package.xml#L15

Can you give an example on this:  "provide a link demonstrating that they are authorized maintainers/releasers of these ROS packages." ? 

@Nir-Az  and I (@SamerKhshiboun) are part of the IntelRealSense Organization

**Comment by SamerKhshiboun** [comment]
@nuclearsandwich, any update on this?
Should I edit/add something ?

File: realsense.tf (hunk 2 above)
Comment (nuclearsandwich) [reply realsense.tf:9]: > Renaming of the old release repositories. The new ones links:

I think that we may be miscommunicating, [IntelRealSense/librealsense2-release](https://github.com/IntelRealSense/librealsense2-release) does not appear to have any common commits from the `master` branch (bloom's central config branch) of [ros2-gbp/librealsense-release](https://github.com/ros2-gbp/librealsense-release) which suggests to me that this is a new release repository and not an extension of [ros2-gbp/librealsense-release](https://github.com/ros2-gbp/librealsense-release).

The same is the case for [ros2-gbp/ros2_intel_realsense-release](https://github.com/ros2-gbp/ros2_intel_realsense-release) and [IntelRealSense/realsense-ros-release](https://github.com/IntelRealSense/realsense-ros-release)

I think the right thing to do is:

1. Archive both
* ros2-gbp/librealsense-release
* ros2-gbp/ros2_intel_realsense-release

2. and then create new ros2-gbp repositories:
* ros2-gbp/librealsense2-release
* ros2-gbp/realsense-ros-release

3. and mirror push the current contents of the IntelRealSense hosted repositories:
* https://github.com/IntelRealSense/librealsense2-release -> https://github.com/ros2-gbp/librealsense2-release
* https://github.com/IntelRealSense/realsense-ros-release -> https://github.com/ros2-gbp/realsense-ros-release

4. finally, once we've verified that you can push releases to the ros2-gbp repositories it's a good idea to archive the ones in IntelRealSense to avoid accidentally pushing to them in the future.


If that plan works for you, let me know and I'll refactor this PR to include the archivals and mirror the current repositories.

Comment (SamerKhshiboun) [reply realsense.tf:9]: Hi @nuclearsandwich,

Thanks for the detailed solution you are proposing.
Before doing the steps above, I need some clarifications from your side:
- How this is going to affect current/old release for the different distros ? (humble, iron, and even EOL distro.. foxy, galactic...)
- What about writ access to gbp repo ? will we have the permissions to write/edit patches and files of our releases ?
- How does this change will affect our future releases ? specially when new ROS distros are out 

Thanks,
Samer

Comment (nuclearsandwich) [reply realsense.tf:9]: > How this is going to affect current/old release for the different distros ? (humble, iron, and even EOL distro.. foxy, galactic...)

Archiving the repositories will make them read-only but keep them available for people using those earlier packages.

> What about writ access to gbp repo ? will we have the permissions to write/edit patches and files of our releases ?

Release repositories are generally _not_ modified by hand, the `bloom` release tool is primarily used to update release repositories based on changes in the upstream source repository.

> How does this change will affect our future releases ? specially when new ROS distros are out

The release process is the same for most ROS 2 packages. Once granted access to the ros2-gbp repositories you'll be able to follow http://docs.ros.org/en/rolling/How-To-Guides/Releasing/Subsequent-Releases.html when making releases.

**Comment by Nir-Az** [comment]
There are 3 issues in the process that will make our lives hard.

1. When a brach is changed we saw the bloom release is not modifing it, so we puch the change manually.

2. When we publish a release we sometimes need control on patches, delete some, create some.
It probably be harder doing it this way.

Any known work method to get by this?

**Comment by nuclearsandwich** [comment]
Okay, what's going on with these is actually really hard to follow. So I'm going to try and spell it all out.

The source repositories to be released are:
1. https://github.com/IntelRealSense/librealsense
2. https://github.com/IntelRealSense/realsense-ros

There are two existing release repositories in `ros2-gbp`
A. https://github.com/ros2-gbp/librealsense-release which was previously used to release https://github.com/IntelRealSense/librealsense
B. https://github.com/ros2-gbp/ros2_intel_realsense-release which was previously used to release https://github.com/intel/ros2_intel_realsense (now archived upstream)

The two external release repositories proposed for import are:
i. https://github.com/IntelRealSense/librealsense2-release for https://github.com/IntelRealSense/librealsense
ii. https://github.com/IntelRealSense/realsense-ros-release for  https://github.com/IntelRealSense/realsense-ros

Both [A] and [i] use [1] as their upstream but have completely distinct release artifacts.
The upstream source repository for [B] has been archived and is apparently succeeded by [2].

So the proposal I mentioned in https://github.com/ros2-gbp/ros2-gbp-github-org/pull/297#discussion_r1278175639 is still valid:
1. Archive both [A] and [B] in ros2-gbp.
2. Mirror push [i] and [ii] into the ros2-gbp organization
3. Archive the external repositories https://github.com/IntelRealsense/librealsense2-release and https://github.com/IntelRealsense/realsense-ros-release to prevent accidental pushes to them instead of the new ros2-gbp repositories).

The questions raised in that thread are answered in https://github.com/ros2-gbp/ros2-gbp-github-org/pull/297#discussion_r1290926134 but I'm happy to go into more detail if there are follow-up questions.



[1]: https://github.com/IntelRealSense/librealsense
[2]: https://github.com/IntelRealSense/realsense-ros
[A]: https://github.com/ros2-gbp/librealsense-release
[B]: https://github.com/ros2-gbp/ros2_intel_realsense-release
[i]: https://github.com/IntelRealSense/librealsense2-release
[ii]: https://github.com/IntelRealSense/realsense-ros-release

---

Separately, there is the issue of maintainership evidence. There is no formal set of pass/fail logic for demonstrating maintainership. "claiming" a reference to a maintainer entry in package.xml is usually enough that if access is ever questioned we can point to the link as evidence. Other links such as PRs merged by the specified GitHub user (merged by in order to demonstrate write access, not authored by) or endorsement from an existing maintainer would both be acceptable substitutes.

**Comment by nuclearsandwich** [comment]
> There are 3 issues in the process that will make our lives hard.

Just to make sure I am not misunderstanding, I only see two points below this sentence not three.



> 1. When a branch is changed we saw the bloom release is not modifying it, so we push the change manually.

Are you pushing changes to the source repository and not seeing those changes reflected in Bloom? Bloom is a tool for generating platform-specific packaging information for releases. Which means that it will look for tagged releases in the source repository. Updating a branch without creating a new release would not change what bloom creates release information for.
The steps [Updating changelog](https://docs.ros.org/en/rolling/How-To-Guides/Releasing/Subsequent-Releases.html#updating-changelog) and [Bump the package version](https://docs.ros.org/en/rolling/How-To-Guides/Releasing/Subsequent-Releases.html#bump-the-package-version) cover the usual method for performing these operations on ROS packages but you can do it manually as well.


> 2\. When we publish a release we sometimes need control on patches, delete some, create some.
>         It probably be harder doing it this way.

What kinds of patching are you talking about? Are you referring to updates to the release once it is already tagged or platform / architecture-specific patches that need to be applied during the release?

**Comment by Nir-Az** [comment]
> > There are 3 issues in the process that will make our lives hard.
> 
> Just to make sure I am not misunderstanding, I only see two points below this sentence not three.
> 
> > 1. When a branch is changed we saw the bloom release is not modifying it, so we push the change manually.
> 
> Are you pushing changes to the source repository and not seeing those changes reflected in Bloom? Bloom is a tool for generating platform-specific packaging information for releases. Which means that it will look for tagged releases in the source repository. Updating a branch without creating a new release would not change what bloom creates release information for. The steps [Updating changelog](https://docs.ros.org/en/rolling/How-To-Guides/Releasing/Subsequent-Releases.html#updating-changelog) and [Bump the package version](https://docs.ros.org/en/rolling/How-To-Guides/Releasing/Subsequent-Releases.html#bump-the-package-version) cover the usual method for performing these operations on ROS packages but you can do it manually as well.
> 
> > 2. When we publish a release we sometimes need control on patches, delete some, create some.
> > It probably be harder doing it this way.
> 
> What kinds of patching are you talking about? Are you referring to updates to the release once it is already tagged or platform / architecture-specific patches that need to be applied during the release?

I will try to explain,

1. An example to our patches is this: https://github.com/IntelRealSense/librealsense2-release/blob/patches/release/humble/librealsense2/0001-update-lrs-options.patch

We need control over this patches, modify/delete/add
The package source code is : https://github.com/IntelRealSense/librealsense and it is cross compile SDK.
For ROS we need to override some code (mostly change some CMake flags)

How can we do it with the new process?

2. We had to release from a different branch and this was missed on bloom release process.
We had to manually push a commit to fix it (https://github.com/ros/rosdistro/commit/a080a91baa3ef72665314eadd775e13e1c4093b2)
Is this a known bug? or did we do something wrong?

Thanks!

**Comment by nuclearsandwich** [comment]
> An example to our patches is this: https://github.com/IntelRealSense/librealsense2-release/blob/patches/release/humble/librealsense2/0001-update-lrs-options.patch
> 
> We need control over this patches, modify/delete/add

Thanks for the context. That patch looks like it was created using bloom's built-in patching workflow, which is definitely under-documented, but referenced in the tutorial here: http://wiki.ros.org/bloom/Tutorials/ReleaseThirdParty#Adding_an_Install_Rule_as_a_Patch

> How can we do it with the new process?

This patch is introduced via https://github.com/IntelRealSense/librealsense2-release/commit/f1d1af2e432b20b9a4d43523ecd13a80d2df2346 and subsequent patches can be added the same way using the same workflow when using the ros2-gbp repository. Your ability to update these patches is not affected by updating the release repository location.

> We had to release from a different branch and this was missed on bloom release process.
> We had to manually push a commit to fix it ([ros/rosdistro@a080a91](https://github.com/ros/rosdistro/commit/a080a91baa3ef72665314eadd775e13e1c4093b2))
> Is this a known bug? or did we do something wrong?

Bloom is configured by the `tracks.yaml` file on the `master` branch. The [rolling](https://github.com/IntelRealSense/librealsense2-release/blob/e7bbc396834a589c0e5f69e22223722c736fd0e2/tracks.yaml#L245-L271) stanza for librealsense2-release has no `devel_branch` specified. Bloom will check the devel_branch for tags when automating a bloom release. However, the release [version](https://github.com/IntelRealSense/librealsense2-release/blob/e7bbc396834a589c0e5f69e22223722c736fd0e2/tracks.yaml#L271) in your rolling track is currently hard-coded to `2.54.1` so no matter which branch is specified the tag corresponding to that release will be used until the tracks.yaml is updated.

Bloom releases are run from source repository tags only, not branches. Bloom can use source branches to find the appropriate tag for a release but your repository is not set up for this currently. https://github.com/ros/rosdistro/commit/a080a91baa3ef72665314eadd775e13e1c4093b2 updates the rosdistro index to specify which branch should be considered for upstream source and documentation builds. However, bloom does not actually use those values (https://github.com/ros-infrastructure/bloom/issues/708) and instead looks exclusively at the branch specified by `devel_branch` if the release version for the track is set to `:{auto}`. With the hard-coded version field you'll have to use `bloom-release --edit ...` whenever you want to change the version you intend to release. That is also a valid workflow.

**Comment by SamerKhshiboun** [comment]
@kadiredd
Please take a moment to read the whole thread and comment here when we are ready to do the requested steps.
We should just make sure that the current releases in humble/iron will not be affected when we mirror our release repositors.

The steps are well-explained above by @nuclearsandwich :
https://github.com/ros2-gbp/ros2-gbp-github-org/pull/297#issuecomment-1694096614

**Comment by SamerKhshiboun** [comment]
Hi @nuclearsandwich, @kadiredd is a ROS Wrapper for RealSense cameras developer, and will continue this task from here. 

Just a quick question, can we set the maintainer as `Librealsense ROS Team` ? or should it be a real GitHub user ?
if that is okay, @kadiredd will submit a new PR for changing the realsense team.

In all of our three packages this is the maintainer:
`<maintainer email="librs.ros@intel.com">LibRealSense ROS Team</maintainer>`

Attached links to the 3 packages xmls

1. https://github.com/IntelRealSense/realsense-ros/blob/ros2-development/realsense2_description/package.xml
2. https://github.com/IntelRealSense/realsense-ros/blob/ros2-development/realsense2_camera_msgs/package.xml
3. https://github.com/IntelRealSense/realsense-ros/blob/ros2-development/realsense2_camera/package.xml

Thanks,
Samer

**Comment by nuclearsandwich** [comment]
> Hi @nuclearsandwich, @kadiredd is a ROS Wrapper for RealSense cameras developer, and will continue this task from here.

Hello @kadiredd, I'm really happy to get this all sorted out. If it will help to get on a video chat once you've made your first pass through the thread. I'm in the US/Pacific timezone but can accommodate an early morning or later evening if required.

**Comment by kadiredd** [comment]
Hello Steven (@nuclearsandwich),
              Thanks for your response and offering live meeting!
I am looking for your email contact and I can’t find one in repo page or elsewhere. May I request you to send us your email contact for scheduling a live meeting invitation? And also, please let us your convenient time for having this meeting, probably on Monday (24th June) morning at/after 8AM your time, works for you?

Regards,
Madhu

**Comment by kadiredd** [comment]
@nuclearsandwich - friendly reminder

**Comment by ashrafk93** [comment]
Hi @nuclearsandwich,

My name is Ashraf, and I am a developer and maintainer for Intel RealSense. I would like to take over handling this thread.

I've reviewed all the comments and relevant documentation, and I believe the proposal you suggested is the right direction to proceed.

Just to confirm my understanding: archiving [ros2-gbp/librealsense-release](https://github.com/ros2-gbp/librealsense-release) would prevent us from making new releases to that repository stream, but the history of previous releases against the ROS distribution would be preserved.
and the ROS build farm will continue to serve the previously built packages from this release repo.

once we mirror the intel org release repos to the newly created repos under ros-gbp org then we would be able to create all our bloom-release for all the repos we specify in this PR i.e
ros-gbp/realsense2-ros-release and ros-gbp/realsense2-release

 I’ve opened a new issue using the [update membership template](https://github.com/ros2-gbp/ros2-gbp-github-org/issues/725) as first step of this transition.
 
 Also once the membership issue is approved and the two new release repos are created, we might need to close this PR and open new 1 with the correct changes.
 
 questions:
 how do i archive old  [ros2-gbp/librealsense-release](https://github.com/ros2-gbp/librealsense-release) ?
can u help us create the new ros2-gbp repos?
 

please let me know if there anything else i can do to proceed with this thread.

**Comment by ashrafk93** [comment]
can someone please review this?
@ChrisThrasher @nuclearsandwich

**Comment by nuclearsandwich** [comment]
> Just to confirm my understanding: archiving [ros2-gbp/librealsense-release](https://github.com/ros2-gbp/librealsense-release) would prevent us from making new releases to that repository stream, but the history of previous releases against the ROS distribution would be preserved.

This is correct.

> and the ROS build farm will continue to serve the previously built packages from this release repo.

Older rosdistro entries that reference this release repository would continue to do so. The binary packages are not hosted in this release repository but in the ROS apt and rpm repositories on packages.ros.org. These repositories host binaries that are in the latest sync of each rosdistro. For older rosdistros that still contain the librealsense package, it will be included there and archiving the release repository will have no effect on that.

> how do i archive old [ros2-gbp/librealsense-release](https://github.com/ros2-gbp/librealsense-release) ?
> can u help us create the new ros2-gbp repos?

This will be handled by an update to this PR listing the librealsense-release repository as archived.

File: realsense.tf (hunk 2 above)
Comment (Nir-Az) [reply realsense.tf:9]: @nuclearsandwich can we please proceed with this PR?

**Review by Nir-Az — APPROVED** [review]
LGTM

**Comment by Nir-Az** [comment]
@nuclearsandwich are we good to proceed with merging this PR?

**Comment by nuclearsandwich** [comment]
This change has been deployed. I had to remove IntelRealsense from the team list because it's an org not an individual.
